### ISSUE TITLE
Subproducts not returned by module API REST

### ISSUE THREAD

**Issue by QuentinPetel** [Dolibarr/dolibarr#19373]
# Instructions

# Bug
When trying to get subproducts through the REST API, false is returned instead of the actual subproducts.

## Environment
- **Version**: 14.0.3
- **OS**: *
- **Web server**: *
- **PHP**: *
- **Database**: *
- **URL(s)**: /api/index.php/explorer/#!/products/productsRetrieveCategories

## Expected and actual behavior
Actual behaviour : 
```
[
  false,
  false
]
```

ExpectedBehaviour :
```
[
  myFirstSubproduct,
  mySecondSubproduct,
]
```
  
## Steps to reproduce the behavior

Use version 14.0.3 and try to use the Rest API to get products and subproducts (dans l'onglet Kits d'un produit)
 

## [Attached files](https://help.github.com/articles/issue-attachments) (Screenshots, screencasts, dolibarr.log, debugging informations…)
![Issue subproducts](https://i.ibb.co/XkfSqR6/Issue-dolibarr.png)

**Comment by atm-maxime** [comment]
It should have been fixed by #19126.
Can you check you are using the last 14.0.3 from GitHub ?

**Comment by QuentinPetel** [comment]
It seems so :

![5927BCCC-2D39-4441-8A9A-F0C18E32A284](https://user-images.githubusercontent.com/16292377/141652056-3f8cfdf8-a3a8-4d49-82d1-7e70940ee080.png)

I actually use the upshift/dolibarr [Dockerfile](https://github.com/upshift-docker/dolibarr/blob/master/Dockerfile) to build my image. Is there multiple 14.0.3 archive published ?

**Comment by QuentinPetel** [comment]
After rebuilding with --no-cache, same issue. It seems that the fix is not embedded in the [14.0.3](https://github.com/Dolibarr/dolibarr/blob/14.0.3/htdocs/product/class/api_products.class.php#L465) version :

```php
                $keys = ['rowid', 'qty', 'fk_product_type', 'label', 'incdec'];
                $childs = [];                                           
                foreach ($childsArbo as $values) {
                        $childs[] = array_combine($keys, $values);
                }                  
```

**Comment by ksar-ksar** [comment]
Hello,

It has been integrated on V14.0.4 : https://github.com/Dolibarr/dolibarr/blob/14.0.4/htdocs/product/class/api_products.class.php#L465
